server/src/main/java/org/elasticsearch/search/vectors/IVFKnnFloatVectorQuery.java

```diff
@@ -72,28 +75,45 @@ public int hashCode() {
         return result;
     }
 
+    @Override
+    VectorScorer createVectorScorer(LeafReaderContext context, FieldInfo fi) throws IOException {
+        LeafReader reader = context.reader();
+        FloatVectorValues vectorValues = reader.getFloatVectorValues(field);
+        if (vectorValues == null) {
+            FloatVectorValues.checkField(reader, field);
+            return null;
+        }
+        return vectorValues.scorer(query);
```

Comment:
What's the difference on doing this wrt just get the number of vectors using vectorValues#size?